playbooks/cluster.yml

```
@@ -52,6 +52,7 @@
     - { role: kubernetes/kubeadm, tags: kubeadm}
     - { role: kubernetes/node-label, tags: node-label }
     - { role: kubernetes/node-taint, tags: node-taint }
+    - { role: kubernetes-apps/gateway_api, when: gateway_api_enabled, tags: gateway_api }
```

I think you need to have some `delegate_to`: first_control_plane here because AFAICS the role use download without limiter, which means you would download the manifests on every node.

---

There seems to be another error, but I added conditions to the download.

```
TASK [kubernetes-apps/gateway_api : Download_file | Copy file from cache to nodes, if it is available] ***
changed: [kub-1]
FAILED - RETRYING: [kub-2 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (4 retries left).
FAILED - RETRYING: [kub-3 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (4 retries left).
FAILED - RETRYING: [kub-2 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (3 retries left).
FAILED - RETRYING: [kub-3 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (3 retries left).
FAILED - RETRYING: [kub-2 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (2 retries left).
FAILED - RETRYING: [kub-3 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (2 retries left).
FAILED - RETRYING: [kub-2 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (1 retries left).
FAILED - RETRYING: [kub-3 -> kub-1]: Download_file | Copy file from cache to nodes, if it is available (1 retries left).
fatal: [kub-2 -> kub-1(10.211.55.22)]: FAILED! => {"attempts": 4, "changed": false, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --rsh='/usr/bin/ssh -S n
one -i /Users/jerry/.vagrant.d/insecure_private_key -o Port=22 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o IdentitiesOnl
y=yes -o IdentityFile=/Users/jerry/.vagrant.d/insecure_private_keys/vagrant.key.ed25519 -o IdentityFile=/Users/jerry/.vagrant.d/insecure_private_keys/vagrant.key.rsa -o Co
ntrolMaster=auto -o ControlPersist=60s' --out-format='<<CHANGED>>%i %n%L' /Users/jerry/kubespray_cache/gateway-api-standard-install.yaml vagrant@10.211.55.24:/tmp/releases
/gateway-api-standard-install.yaml", "msg": "Warning: Identity file /Users/jerry/.vagrant.d/insecure_private_key not accessible: No such file or directory.\nWarning: Perma
nently added '10.211.55.24' (ED25519) to the list of known hosts.\r\nno such identity: /Users/jerry/.vagrant.d/insecure_private_keys/vagrant.key.ed25519: No such file or d
irectory\r\nno such identity: /Users/jerry/.vagrant.d/insecure_private_keys/vagrant.key.rsa: No such file or directory\r\nPermission denied, please try again.\r\nPermissio
n denied, please try again.\r\nvagrant@10.211.55.24: Permission denied (publickey,password).\r\nrsync: connection unexpectedly closed (0 bytes received so far) [sender]\nr
sync error: unexplained error (code 255) at io.c(232) [sender=3.2.7]\n", "rc": 255}
```

---

Sorry I wasn't clear, I meant
```suggestion
    - role: kubernetes-apps/gateway_api
     when: gateway_api_enabledgateway_api
     delegate_to: groups['kube_control_plane'][0]
     run_once: true
```

Though it it interferes with delegate_localhost, we can probably instead add a when instead at the group level.
But no need to have it on every task of the role instead.

---

I think I didn't add `run_once`; it looks like it's working.